core/src/com/treode/async/misc/private/DirectDeque.scala

```diff
@@ -0,0 +1,92 @@
+package com.treode.async.misc
+
+import java.lang.System
+import scala.reflect.ClassTag
+
+/**
+ * DirectDeque is a queue (FIFO ordering) that supports
+ * constant time access to any element in the queue
+ */
+class DirectDeque [M] (implicit
+     mtag: ClassTag[M]
+) {
+     private val init_capacity = 16
+     private var front = 0
+     private var back = front
+     private var _size = 0
+
+     private var array = new Array[M](init_capacity)
+
+     def size() = _size
```

Comment:
I think this compute rather than track size. It will need to be computed two different ways, depending on if front<=back or not.